packages/x-date-pickers/src/internals/components/CalendarOrClockPicker/CalendarOrClockPicker.tsx

```diff
@@ -158,6 +173,15 @@ export function CalendarOrClockPicker<TDate, View extends CalendarOrClockPickerV
           toggleMobileKeyboardView={toggleMobileKeyboardView}
         />
       )}
+      {showTabs && (
```

Comment:
We can avoid creating an empty component above by adding a check here

```suggestion
      {showTabs && !!TabsComponent && (
```